SkietbaanBE/SkietbaanBE/Controllers/UserController.cs

```diff
@@ -10,50 +14,63 @@ namespace SkietbaanBE.Controllers
     public class UserController : Controller
     {
         private ModelsContext _context;
-        public UserController(ModelsContext db) {
+        public UserController(ModelsContext db)
+        {
             _context = db;
         }
-
-        //public UserController() { }
-
         // GET: api/User
         [HttpGet]
-        public IEnumerable<User> Get() {
-            var users = _context.Users.ToArray<User>();
-            return users;
+        public IEnumerable<User> Get()
+        {
+           return _context.Users.ToArray<User>();
         }
-
         // GET: api/User/5
         [HttpGet("{id}", Name = "Get")]
-        public User Get(int id) {
-            User user = _context.Users.Find(id);
-            return user;
-        }
-
-        [HttpGet("test/{ret}", Name = "Test")]
-        public string Test(string ret) {
-            return $"Returned : {ret}";
+        public async Task<User> Get(int id)
+        {
+            return await _context.Users.FindAsync(id);
         }
-
         // POST: api/User
-        [HttpPost]
-        public virtual void Post([FromBody] User user) {
-            _context.Add(user);
-            _context.SaveChanges();
+        [HttpPost("{id}")]
+        public async Task<HttpResponseMessage> Post(int id,[FromBody] User user)
+        {
+            if (ModelState.IsValid)
+            {
+                //get user with the specified ID from database
+                User dbUser = await _context.Users.FindAsync(id);
+                //user not found
+                if(dbUser == null)
+                {
+                    return new HttpResponseMessage(HttpStatusCode.NotFound);
+                }
+                //get today's date and save it under user entry date
+                user.EntryDate = DateTime.Now;
+                //Save User
+                await _context.AddAsync(user);
+                await _context.SaveChangesAsync();
+                return new HttpResponseMessage(HttpStatusCode.OK);
+            }
+            else
+            {
+                return new HttpResponseMessage(HttpStatusCode.BadRequest);
+            }
         }
-
-        // PUT: api/User/5
+        // PUT: api/User/
         [HttpPut]
-        public void Put([FromBody] User user) {
+        public async Task<ActionResult> Put([FromBody] User user)
+        {
+            //error handling, check if client provided valid data
+            if (user == null)
+            {
+                return new BadRequestObjectResult("user cannot be null");
+            }
+            else if (Get(user.Id) == null)
+            {
+                return new OkObjectResult("user does not exist");
```

Comment:
Better to `return NotFound("User does not exist");` here

Reply:
working on the changes now, i had a hard time trying to figure out which to use

Reply:
Cool, feel free to DM if you still don't understand